packages/runner-shared/src/event-manager.js

```diff
@@ -126,10 +127,17 @@ export const eventManager = {
 
     _.each(socketToDriverEvents, (event) => {
       ws.on(event, (...args) => {
-        Cypress.emit(event, ...args)
+        // these events are set up before Cypress is instantiated, so it's
+        // possible it's undefined when an event fires, but it's okay to
+        // ignore at that point
+        Cypress?.emit(event, ...args)
```

Comment:
Are we sure these are ok to ignore? We could also queue them up and fire them on connection.

Reply:
I don't know if this is even necessary. I'm going to revert this and, if all tests pass, assume it was a non-issue.

Reply:
Updated [3e6d6bf](https://github.com/cypress-io/cypress/pull/18075/commits/3e6d6bfe155234f782a890fd904b361c84d5c34e)